extensions/legit/porcelain-git.lisp

```diff
@@ -0,0 +1,416 @@
+(uiop:define-package :lem/porcelain-git
+  (:use :cl :lem/porcelain)
+  (:shadow :push)
+  (:import-from :trivial-types
+                :proper-list)
+  (:import-from :lem/porcelain
+                :porcelain-error)
+  (:export :git-project-p)
+  (:documentation "Implements the porcelain interface for git-based repos."))
+
+(in-package :lem/porcelain-git)
+
+(declaim (type (proper-list string) *git-base-arglist*))
+(defvar *git-base-arglist* (list "git")
+  "The git program, to be appended command-line options.")
+
+(defvar *branch-sort-by* "-creatordate"
+  "When listing branches, sort by this field name.
+  Prefix with \"-\" to sort in descending order.
+
+  Defaults to \"-creatordate\", to list the latest used branches first.")
+
+(defvar *file-diff-args* (list "diff" "--no-color")
+  "Arguments to display the file diffs.
+  Will be surrounded by the git binary and the file path.
+
+  For staged files, --cached is added by the command.")
+
+(defvar *verbose* nil)
+
+;; VCS implementation for git
+(defclass vcs-git () 
+  ((rebase-pid :type (or nil string)
```

Comment:
This is clearly good as it removes a current limiting factor, thanks for this, so it would have benefited from a separate PR, as the related changes are not visible in the diff and this change doesn't imply all the others.

Reply:
Couldn't we maintain a HT of project-root -> PID? Avoiding an object :D  

Shall we store more data into a VCS object?  (They would only be singletons, only one VCS object per project, right?)

Reply:
Yeah, my idea was to eventually shove more state into various `vcs`objects: I was thinking for `libgit2` specifically, but it might be useful in the future for other implementations as well

Reply:
I think we can store more data ino the VCS object, but that's for later.

Maintaining a hash-table of project root -> rebase PID is trying to avoid OO very much^^ not a great idea.

We can close this conversation.